src/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -99,16 +99,52 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (offset < 0 || offset > buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if (count < 0 || count > (buffer.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
             _buffer.Write(buffer, offset, count);
         }
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
```

Comment:
We usually also write explicit tests to validate this as well making sure that there is no Task returned by rather an exception directly.  This is difficult to test if you use "await" logic because it is unclear whether the exception is returned directly or via a Task since "await" will blur that distinction.  So, it's important that the tests are written using Assert.* making sure no Task is returned.

cc: @stephentoub